### PR TITLE
"cudnn operators change to cudnn kernel"

### PR DESCRIPTION
**Pull request by dzhwinter**
fix https://github.com/PaddlePaddle/Paddle/issues/6995

### CODE REVIEW

File: python/paddle/v2/fluid/tests/test_conv2d_op.py [hunk 1]
```diff
@@ -113,93 +114,79 @@ def init_op_type(self):
...
-        core.use_cudnn()
-        self.op_type = "conv2d_cudnn"
-
-
-class TestCudnnWithStride(TestWithStride):
-    def init_op_type(self):
-        core.use_cudnn()
-        self.op_type = "conv2d_cudnn"
-
+# class TestWithPad(TestConv2dOp):
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/tests/test_conv2d_op.py]: 为啥都注释掉了？

File: paddle/framework/data_transform.cc [hunk 2]
```diff
@@ -19,16 +19,19 @@ limitations under the License. */
...
 namespace framework {
 
-Tensor* DataTransform(const OpKernelType& expected_kernel_type,
-                      const OpKernelType& kernel_type_for_var,
-                      const Tensor& input_tensor) {
-  Tensor* out = nullptr;
+void DataTransform(const OpKernelType& expected_kernel_type,
+                   const OpKernelType& kernel_type_for_var,
+                   const Tensor& input_tensor, Tensor* out) {
+  if (expected_kernel_type.library_type_ != kernel_type_for_var.library_type_) {
```
Comment (jacquesqiao) [inline paddle/framework/data_transform.cc]: out->set_library()?

Comment (dzhwinter) [reply paddle/framework/data_transform.cc]: No, library is a mark of a kernel. Tensor does not have that field.

File: paddle/framework/data_transform.cc [hunk 3]
```diff
@@ -19,16 +19,19 @@ limitations under the License. */
...
+    Copy(input_tensor, expected_kernel_type.place_, out);
```
Comment (jacquesqiao) [inline paddle/framework/data_transform.cc]: 突然在想，library需要参与transform运算么，还是只需要参与选kernel就行了

Comment (dzhwinter) [reply paddle/framework/data_transform.cc]: yes. only the mkldnn need to do the transform.

**Pull request merged**